server/controllers/systemController.js

```diff
@@ -29,15 +32,117 @@ const SystemController = () => {
   })
 
   router.get('/sync', async (req, res) => {
+    const params = new URLSearchParams();
+    params.append('grant_type', 'client_credentials');
+    params.append('client_id', 'hyperloot');
+    params.append('client_secret', 'ef8bfd1fb62ecca4354ab6f4f1852cd986831b79b97e5e3478');
+
+    let auth = await fetch('https://api.abiosgaming.com/v2/oauth/access_token', {
+      method: 'POST',
+      body: params,
+    });
 
-    // Some good code to sync all matches and tournaments
-    // const matches = await MatchModel.find()
-    const matches = await MatchModel.update({ endDate: { $lte: Date.now() } }, { completed: true }, { multi: true })
+    auth = await auth.json();
+    const token = auth.access_token;
 
-    res.send({
-      matches
-    })
+    const getAbios = (endPoint, params) => {
+      const url = `https://api.abiosgaming.com/v2/${endPoint}?access_token=${token}`;
+
+      if (params) {
+        const string = Object.entries(params).map(([key, val]) => `${key}=${val}`).join('&');
+        return fetch(`${url}&${string}`);
+      }
+
+      return fetch(url);
+    };
+
+    const loadPaginatedData = async (endPoint, params) => {
+      console.log(`${endPoint} is loading`);
+      let firstPage = await getAbios(endPoint, params);
+      firstPage = await firstPage.json();
+      console.log(`1 of ${firstPage.last_page} page loaded`);
+
+      let list = firstPage.data;
+
+      if (firstPage.last_page === 1) {
+        return list;
+      }
+
+      let promises = [];
+      let timeout = 0;
+
+      for (let i = 2; i < firstPage.last_page + 1; i++) {
+        const advancedParams = Object.assign(params, { page: i });
+
+        const promise = new Promise(resolve => {
```

Comment:
у тебя не последовательно будут выполнятся промисы... потому что ты оборачиваешь в коллбек с задержкой, а потом ВСЕ РАЗОМ запускаешь... смотри, надо написать for обычным (не функциональным стилем) и внутри делать await

```suggestion
        let page = await getAbios(...
        list = list.concat(item);
        await wait(500);
```

где wait будет Promise 
```js
const wait = (time) => new Promise(resolve => setTimeout(() => resolve(), time);
```